### PR TITLE
Hide `SelectableRegion` text selection toolbar when orientation changes

### LINKED ISSUE #104816: `SelectionArea` should hide toolbar when orientation changes on mobile
https://github.com/flutter/flutter/pull/103512 implemented this for `EditableText` but the same should be done for `SelectionArea`. On Android the handles should be hidden as well as the toolbar.

https://user-images.githubusercontent.com/68219924/170675638-e19faa11-b89a-45e7-b4cc-71640eda9b47.mp4

https://user-images.githubusercontent.com/68219924/170675641-0c9f1cf6-37ad-45cd-9a6e-18807d6c7986.mp4

<details>
 <summary>flutter doctor -v</summary>

```
[√] Flutter (Channel master, 3.1.0-0.0.pre.965, on Microsoft Windows [Version 10.0.19044.1706], locale en-GB)
    • Flutter version 3.1.0-0.0.pre.965 at C:\Development\flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 9e7f13c2a4 (42 minutes ago), 2022-05-27 05:08:09 -0400
    • Engine revision 866faabf89
    • Dart version 2.18.0 (build 2.18.0-149.0.dev)
    • DevTools version 2.13.1

[√] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at C:\Users\marku\AppData\Local\Android\sdk
    • Platform android-32, build-tools 32.0.0
    • Java binary at: C:\Users\marku\AppData\Local\JetBrains\Toolbox\apps\AndroidStudio\ch-0\212.5712.43.2112.8609683\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.11.7)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.11.31911.196
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2021.2)
    • Android Studio at C:\Users\marku\AppData\Local\JetBrains\Toolbox\apps\AndroidStudio\ch-0\212.5712.43.2112.8609683
    • Flutter plugin version 67.1.2
    • Dart plugin version 212.5744
    • Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)

[√] Connected device (4 available)
    • sdk gphone64 x86 64 (mobile) • emulator-5554 • android-x64    • Android 12 (API 31) (emulator)
    • Windows (desktop)            • windows       • windows-x64    • Microsoft Windows [Version 10.0.19044.1706]
    • Chrome (web)                 • chrome        • web-javascript • Google Chrome 101.0.4951.67
    • Edge (web)                   • edge          • web-javascript • Microsoft Edge 101.0.1210.47

[√] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```
</details>

<details>
 <summary>Sample</summary>

```dart
import 'package:flutter/material.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  static const String _title = 'Flutter Sample';

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: _title,
      home: SelectionArea(
        child: Scaffold(
          appBar: AppBar(title: const Text(_title)),
          body: const Padding(
            padding: EdgeInsets.all(24.0),
            child: Text('word\nword word word\nword'),
          ),
        ),
      ),
    );
  }
}
```
</details>

cc @chunhtai

### PR DESCRIPTION
**Pull request by markusaksli-nc**
Hide the `SelectableRegion` text selection toolbar on Android and iOS when orientation changes by keeping track of the previous orientation in `didChangeDependencies`. Hide text selection handles as well on Android to match native project behavior.

This also fixes the wrong method call in `hideToolbar`.

Fixes https://github.com/flutter/flutter/issues/104816

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/widgets/selectable_region.dart [hunk 1]
```diff
@@ -228,6 +231,30 @@ class _SelectableRegionState extends State<SelectableRegion> with TextSelectionD
     );
   }
 
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    if (defaultTargetPlatform != TargetPlatform.iOS && defaultTargetPlatform != TargetPlatform.android) {
```
Comment (chunhtai) [inline packages/flutter/lib/src/widgets/selectable_region.dart:237]: Maybe use a switch case that we would catch it if a new platform is added to flutter

File: packages/flutter/lib/src/widgets/selectable_region.dart [hunk 2]
```diff
@@ -228,6 +231,30 @@ class _SelectableRegionState extends State<SelectableRegion> with TextSelectionD
...
+    // Hide the text selection toolbar on mobile when orientation changes.
+    final Orientation orientation = MediaQuery.of(context).orientation;
+    if (_lastOrientation == null) {
+      _lastOrientation = orientation;
+      return;
+    }
+    if (orientation != _lastOrientation) {
+      _lastOrientation = orientation;
+      if (defaultTargetPlatform == TargetPlatform.iOS) {
+        hideToolbar(false);
```
Comment (chunhtai) [inline packages/flutter/lib/src/widgets/selectable_region.dart:250]: The two if cases can be merged to 
hideToolbar(defaultTargetPlatform == TargetPlatform.android)

File: packages/flutter/lib/src/widgets/selectable_region.dart [hunk 3]
```diff
@@ -699,7 +726,7 @@ class _SelectableRegionState extends State<SelectableRegion> with TextSelectionD
   void hideToolbar([bool hideHandles = true]) {
     _selectionOverlay?.hideToolbar();
     if (hideHandles) {
-      _selectionOverlay?.hideToolbar();
+      _selectionOverlay?.hideHandles();
```
Comment (chunhtai) [inline packages/flutter/lib/src/widgets/selectable_region.dart:729]: oops nice catch!

File: packages/flutter/test/widgets/selectable_region_test.dart [hunk 4]
```diff
@@ -1043,6 +1043,46 @@ void main() {
       expect(clipboardData['text'], 'thank');
     }, skip: kIsWeb); // [intended] Web uses its native context menu.
   });
+
+  testWidgets('long press and drag touch selection', (WidgetTester tester) async {
```
Comment (chunhtai) [inline packages/flutter/test/widgets/selectable_region_test.dart:1047]: the test description is inaccurate

**Review by chunhtai — COMMENTED** [review]
Thanks for fixing this issue! The code looks good overall, I left some comments

File: packages/flutter/lib/src/widgets/selectable_region.dart (hunk 1 above)
Comment (jonahwilliams) [inline packages/flutter/lib/src/widgets/selectable_region.dart:237]: drive by comment: here and elsewhere use a `switch` for enums. This helps us track down places to update if we happen to add more platforms in the future 😅

File: packages/flutter/test/widgets/selectable_region_test.dart [hunk 5]
```diff
@@ -1043,6 +1043,46 @@ void main() {
       expect(clipboardData['text'], 'thank');
     }, skip: kIsWeb); // [intended] Web uses its native context menu.
   });
+
+  testWidgets('toolbar hidden on mobile when orientation changes', (WidgetTester tester) async {
```
Comment (chunhtai) [inline packages/flutter/test/widgets/selectable_region_test.dart:1047]:
```suggestion
  testWidgets('toolbar is hidden on mobile when orientation changes', (WidgetTester tester) async {
```

**Review by chunhtai — APPROVED** [review]
LGTM

**Pull request merged**